src/main/java/org/apache/sysds/runtime/util/Py4jConverterUtils.java

```diff
@@ -114,6 +120,82 @@ public static MatrixBlock convertPy4JArrayToMB(byte[] data, int rlen, int clen,
 		return mb;
 	}
 
+	public static Array<?> convert(byte[] data, int numElements, Types.ValueType valueType) {
```

Comment:
we need to add some internal tests (java only) for this function.
however the implementation does look fine. There are some minor improvements possible if we do not cast the array types inside the inner loops but assign the cast arrays outside the loop and then use that allocation.

Reply:
I wasn’t able to find existing tests for the Py4jConverterUtils class. Could you please advise where I should add new tests?

Reply:
I don't think we have any.
You will have to make some from scratch.
It should be fairly simple to allocate byte[] and then call the methods. 

Please put the new tests in 

src/test/java/org/apache/sysds/test/component/frame/array